RobotCode/snobot2018/src/org/snobot/autonomous/AutonomousFactory.java

```diff
@@ -0,0 +1,154 @@
+package org.snobot.autonomous;
+
+import java.io.File;
+
+import org.snobot.Properties2018;
+import org.snobot.SmartDashboardNames;
+import org.snobot.Snobot2018;
+import org.snobot.lib.autonomous.ObservableSendableChooser;
+import org.snobot.lib.autonomous.SnobotAutonCrawler;
+import org.snobot.positioner.IPositioner;
+
+import edu.wpi.first.networktables.NetworkTable;
+import edu.wpi.first.networktables.NetworkTableEntry;
+import edu.wpi.first.networktables.NetworkTableInstance;
+import edu.wpi.first.networktables.NetworkTableValue;
+import edu.wpi.first.networktables.TableEntryListener;
+import edu.wpi.first.wpilibj.command.CommandGroup;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+
+public class AutonomousFactory
+{
+    private static final double sY_START = 336 - 3 * 12;
+
+    protected ObservableSendableChooser<File> mAutonModeChooser;
+    protected ObservableSendableChooser<StartingPositions> mPositionChooser;
+    protected NetworkTable mAutoModeTable;
+
+    protected CommandParser mCommandParser;
+
+    protected IPositioner mPositioner;
+
+    public enum StartingPositions
+    {
+        RedLeft("Red Left", 0, -sY_START, 0), RedMiddle("Red Middle", 0, -sY_START, 0), RedRight("Red Right", 0, -sY_START, 0), BlueLeft("Blue Left",
```

Comment:
We can leave this for now, but these will not be the positions we need for this year.  The field is rotationally symmetrical so we don't need to know if we are red or blue.